### ISSUE TITLE
Notification Feature Suggestion

### ISSUE THREAD

**Issue by nefertiti007** [tananaev/traccar#2061]
Dear Anton & Team,.

I can see the email notification system being built, so was just thinking would the notification recipient email be set per device or per userid.

As per what I feel, per device would be a good idea, when we have groups/and possibility of a single device being under multiple accounts, thus users would be able to specify who to alert for a particular device.

-Sorry wasnt sure if I should post it here or in forum :) so did it here..

**Comment by tananaev** [comment]
User can choose to receive notifications for a device or a group of devices.

**Comment by tananaev** [comment]
Sorry, correction. Geofencing events are per device and groups. Other events are just for all user devices.

@Abyss777, please correct me if I'm wrong.

**Comment by Abyss777** [comment]
@tananaev you are rigth. Geofences are linked to devices and groups, and can be unlinked from user. Other events will be notified for all user devices.

@nefertiti007  Thanks for the useful idea, we will think about how to do that.

**Comment by nefertiti007** [comment]
@Abyss777 & @tananaev Thanks for considering my feedback.. :)

**Comment by tananaev** [comment]
@jaimzj, please explain the use case for this feature.

**Comment by jaimzj** [comment]
The reason why I suggested having the notifications recipients configurable on per device basis are as following.

1. If a single device is assigned to multiple user accounts in current scenario I believe only the useraccount holder will receive notifications,. however allowing per device setting, will allow admin/users to define each device who should get the notifications.

2. Configuring alerts recipients on per device also would allow traccar admins/business users with multiple vehicles to alert the right recipients based on the vehicle/device they are responsible.
- In this case Group will work,. However only if a Single Device does not fall under Multiple Groups.

**Comment by Abyss777** [comment]
This feature is for business installations more I guess.  
1. There is fleet of tracks. There are some operators who see all devices, but they should not get tones of notifications from every device. There are some responsible managers who have to watch their group of devices and get all notifications.
2. There is a boss who want to see everything, but do not want get notifications about everything except his own car. There is some special persons who responsible for every device and only them should get notifications.

May be It looks like crutch, and we should consider deep improvement of device linking and divide it on three types like watch/edit/notifiy and it is a big work.

**Comment by tananaev** [comment]
@jaimzj

1. Every device can decide if he wants to receive notification. Device can be shared between number of people.

2. It's already possible through different user accounts. I don't see much value in adding another option to do the same thing. It would just complicate the server.

3. I don't understand this point.

@Abyss777

1. Makes some sense, but I think in most cases managers would be interested only in their devices.

2. It's possible to have a company account and a personal account with a single car.

I guess it makes sense to allow users to select from which devices they want to receive notifications, but I don't think #2436 is the best solution for this particular problem.

**Comment by Abyss777** [comment]
@tananaev 
You are right #2436 looks like quick fix, we need some more base improvement to choose notification by device/group.
Some analog of linking `device -> user` and `group -> user` but which works for notifications.

**Comment by tananaev** [comment]
Agree. One option would be to extend `user_device` table with a flag.

**Comment by kirantpatil** [comment]
@jaimzj, please correct me if I am wrong.

What I understood from @jaimzj is he wants to assign multiple email recipients to each device.

The recipients may not be the only users registered with traccar but User/Admin can add new email ids as recipients for that device to send notification.

**Comment by tananaev** [comment]
I want to know a use case where non-users would want to receive notifications.

**Comment by djbo1982** [comment]
Hello to all @tananaev @Abyss777  @jaimzj 

Well, in bussines, when we have a clients, and this have another clients, we need to have different email, example

I´m the admin for a group, and i have 10 clients, this 10 clients need to received notification of they devices, but we can config this, because we dont have different place to put different email address.

is a power useful option in the platform and many person go to be happy with this, person like me... hehehhe

**Comment by Abyss777** [comment]
Hello, @djbo1982 

There is better solution for your use case. I would call it User Manager/Management and it will be available in next release.

**Comment by djbo1982** [comment]
@tananaev

I´m reading that new solution…. look very good… and go to solved many problem for us…

I love your work Man… we are so happy for meet your work


El 05/01/2017, a las 23:10, Andrey Kunitsyn <notifications@github.com<mailto:notifications@github.com>> escribió:


Hello, @djbo1982<https://github.com/djbo1982>

There is better solution for your use case. I would call it User Manager/Management and it will be available in next release.

**Comment by tananaev** [comment]
@djbo1982, you should thank @Abyss777 for it. He worked on the implementation. I just reviewed it and gave some suggestions.

**Comment by djbo1982** [comment]
Sure… @Abyss777<https://github.com/Abyss777>

I´m Just changing my installation of Windows server to Ubuntu, i use windows for many years, but this night i decided to change all…

To many problem to install my server to used another system (NOT GPS) but, im ready to begin with UBUNTU…

And yes, this new version have a useful thing for managed float, because now, we don’t have to much to offers to others. and the stability and the develop of traccar let us to implement new thing to offer to the clients…

im learn to how used as well GitHub, because at today, i don’t know how i need to make and change to test your update develop.

Im use Azure from microsoft VPN and Amazon Web Services for all my project

if you need help to test the system, only ask me, i have all time for this


El 05/02/2017, a las 02:17, Anton Tananaev <notifications@github.com<mailto:notifications@github.com>> escribió:


@djbo1982<https://github.com/djbo1982>, you should thank @Abyss777<https://github.com/Abyss777> for it. He worked on the implementation. I just reviewed it and gave some suggestions.
